scaletest/scaletest.sh

```diff
@@ -0,0 +1,182 @@
+SCALETEST_SECRETS_TEMPLATE="${PROJECT_ROOT}/scaletest/terraform/secrets.tfvars.tpl"
+
+log "Writing scaletest secrets to file."
+SCALETEST_NAME="${SCALETEST_NAME}" \
+	SCALETEST_PROJECT="${SCALETEST_PROJECT}" \
+	SCALETEST_PROMETHEUS_REMOTE_WRITE_USER="${SCALETEST_PROMETHEUS_REMOTE_WRITE_USER}" \
+	SCALETEST_PROMETHEUS_REMOTE_WRITE_PASSWORD="${SCALETEST_PROMETHEUS_REMOTE_WRITE_PASSWORD}" \
+	envsubst <"${SCALETEST_SECRETS_TEMPLATE}" >"${SCALETEST_SECRETS}"
+
+pushd "${PROJECT_ROOT}/scaletest/terraform"
+
+echo "Initializing terraform."
+maybedryrun "$DRY_RUN" terraform init
+
+echo "Setting up infrastructure."
+maybedryrun "$DRY_RUN" terraform apply --var-file="${SCALETEST_SCENARIO_VARS}" --var-file="${SCALETEST_SECRETS}" --auto-approve
+
+if [[ "${DRY_RUN}" != 1 ]]; then
+	SCALETEST_CODER_URL=$(<"${CONFIG_DIR}/url")
+else
+	SCALETEST_CODER_URL="http://coder.dryrun.local:3000"
+fi
+KUBECONFIG="${PWD}/.coderv2/${SCALETEST_NAME}-cluster.kubeconfig"
+echo "Waiting for Coder deployment at ${SCALETEST_CODER_URL} to become ready"
+maybedryrun "$DRY_RUN" kubectl --kubeconfig="${KUBECONFIG}" -n "coder-${SCALETEST_NAME}" rollout status deployment/coder
+
+echo "Initializing Coder deployment."
+DRY_RUN="$DRY_RUN" "${PROJECT_ROOT}/scaletest/lib/coder_init.sh" "${SCALETEST_CODER_URL}"
+
+echo "Creating ${SCALETEST_NUM_WORKSPACES} workspaces."
+DRY_RUN="$DRY_RUN" "${PROJECT_ROOT}/scaletest/lib/coder_shim.sh" scaletest create-workspaces \
+	--count "${SCALETEST_NUM_WORKSPACES}" \
+	--template=kubernetes \
+	--concurrency 10 \
+	--no-cleanup
+
+echo "Sleeping 10 minutes to establish a baseline measurement."
+maybedryrun "$DRY_RUN" sleep 600
+
+echo "Sending traffic to workspaces"
+maybedryrun "$DRY_RUN" "${PROJECT_ROOT}/scaletest/lib/coder_workspacetraffic.sh" "${SCALETEST_NAME}"
+maybedryrun "$DRY_RUN" kubectl --kubeconfig="${KUBECONFIG}" -n "coder-${SCALETEST_NAME}" wait pods coder-scaletest-workspace-traffic --for condition=Ready
+maybedryrun "$DRY_RUN" kubectl --kubeconfig="${KUBECONFIG}" -n "coder-${SCALETEST_NAME}" logs -f pod/coder-scaletest-workspace-traffic
+
+echo "Starting pprof"
+maybedryrun "$DRY_RUN" kubectl -n "coder-${SCALETEST_NAME}" port-forward deployment/coder 6061:6060 &
+pfpid=$!
+maybedryrun "$DRY_RUN" trap 'kill $pfpid' EXIT
+
+echo "Waiting for pprof endpoint to become available"
+pprof_attempt_counter=0
+while ! maybedryrun "$DRY_RUN" timeout 1 bash -c "echo > /dev/tcp/localhost/6061"; do
+	if [[ $pprof_attempt_counter -eq 10 ]]; then
+		echo
+		echo "pprof failed to become ready in time!"
+		exit 1
+	fi
+	maybedryrun "$DRY_RUN" sleep 3
+done
+
+echo "Taking pprof snapshots"
+maybedryrun "$DRY_RUN" curl --silent --fail --output "${SCALETEST_NAME}-heap.pprof.gz" http://localhost:6061/debug/pprof/heap
+maybedryrun "$DRY_RUN" curl --silent --fail --output "${SCALETEST_NAME}-goroutine.pprof.gz" http://localhost:6061/debug/pprof/goroutine
+maybedryrun "$DRY_RUN" kill $pfpid
```

Comment:
Suggestion: Could remove trap handler here to avoid error on exit due to missing pid:
```suggestion
maybedryrun "$DRY_RUN" kill $pfpid
maybedryrun "$DRY_RUN" trap - EXIT
```